pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -213,9 +230,10 @@ func (p *Provider) getVnetInfoLabels(subnetID string) (map[string]string, error)
 		return nil, err
 	}
 	vnetLabels := map[string]string{
-		vnetSubnetNameLabel:     vnetSubnetComponents.SubnetName,
-		vnetGUIDLabel:           p.vnetGUID,
-		vnetPodNetworkTypeLabel: consts.NetworkPluginModeOverlay,
+		subnetNameLabel:      vnetSubnetComponents.SubnetName,
+		vnetGUIDLabel:        p.vnetGUID,
+		azureCNIOverlayLabel: strconv.FormatBool(true),
```

Comment:
We only support overlay atm, so this should already be true?

Reply:
We support non-overlay as well

Reply:
This is ripe for refactoring, sometime very soon. Yes, getVnetInfoLabels is currently only called in the context of Azure CNI Overlay, but it is not obvious within the function.